test/Test.svelte

```diff
@@ -0,0 +1,36 @@
+<script lang="ts">
```

Comment:
Just FYI, constructing a fully functional component like this would constitute an integration test. They are good to have, but should be used sparingly given the amount of setup required. Which can be very ridid.

I always recommend the standard and favor lots of small unit tests if and when we can:
https://www.headspin.io/blog/the-testing-pyramid-simplified-for-one-and-all

Given the "hooks" are essentially functions, this should lend itself well to unit testing.

Reply:
Like the location of the tests. I also followed the tests other ports were using. They create a single component that takes in all the props possible. This makes it super reusable. So this `Test.svelte` will be the only component needed.
We sadly can't test without a component since the hook needs a floating and reference element to function.

Reply:
Why not create your own elements? Testing Library likely has a convenience method for this as well:
https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement

Remember, unit tests are not testing the full feature in a real world scenario. They're testing a portion (a unit) of the possible combinations of settings and logic to ensure they behave as expected. That's what separates them from integration/e2e. You purposely mock/spy/stage anything surrounding the small unit you're testing.

Reply:
Great tip! Doing this now which is much cleaner!